lib/internal/modules/esm/resolve.js

```diff
@@ -846,12 +846,13 @@ function moduleResolve(specifier, base, conditions, preserveSymlinks) {
   } else if (protocol === 'file:' && specifier[0] === '#') {
     resolved = packageImportsResolve(specifier, base, conditions);
   } else {
-    try {
-      resolved = new URL(specifier);
-    } catch (cause) {
+    const url = URL.parse(specifier);
```

Comment:
I empathize with the desire not to throw, but `URL.parse` is deprecated: https://nodejs.org/docs/latest/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost

Is there another way this can be achieved?

Edit: Maybe I'm looking at the wrong `url.parse`; are you using the new one? https://nodejs.org/docs/latest/api/url.html#urlparseinput-base

Reply:
I believe this is the new one, since URL is capitalized, and the deprecated one is lowercase.

I found this API since its immediately below the constructor the code was using in lib/internal/url.js. There are no comments on that URL class, the constructor it was using nor the static method I used and its under "internal" so I don't know if this is actually the public facing URL.parse or not, but the name matches the new one and its part of the same class it was using before.